cmd/skopeo/inspect.go

```diff
@@ -235,3 +239,51 @@ func (opts *inspectOptions) writeOutput(stdout io.Writer, data any) error {
 	defer rpt.Flush()
 	return rpt.Execute([]any{data})
 }
+
+func manifestDigestFromManifest(manifestBlob []byte, img types.Image, userAlgorithm digest.Algorithm) (digest.Digest, error) {
+	if userAlgorithm != "" {
+		if !userAlgorithm.Available() {
+			return "", fmt.Errorf("digest algorithm %q is not available", userAlgorithm)
+		}
+		return userAlgorithm.FromBytes(manifestBlob), nil
```

Comment:
With this, `--manifest-digest=anything` is incorrect for v2s1, incl. for `--manifest-digest=sha256`. So I think we need something like a `manifest.DigestWithAlgorithm` in c/image first.